features/step_definitions/generic_api.rb

```diff
@@ -8,6 +8,12 @@
   @deed_hash = @deed.to_hash
 end
 
+Given(/^I have valid deed data with <(\d+)> borrowers$/) do |num_borrowers|
+  borrowers = num_borrowers.to_i
+  @deed = Deed.new(borrowers)
+  @deed_hash = @deed.to_hash
+end
+
```

Comment:
This is a good solution I would be tempted to remove the old steps where the number was hardcoded and refactor existing acceptance tests that used the old steps to use this new better test step.